src/index.js

```diff
@@ -39,30 +39,56 @@ loader.pitch = function (request) {
 
   const remainingRequest = JSON.stringify(workerLoader + '!' + request);
 
+  const wrapperBuilder = `
+  var Worker = require(${remainingRequest});
+  var wrap = require('comlink').wrap;
+
+  function getWrappedWorker() {
+    var worker = Worker();
+    var wrapper = wrap(worker);
+
+    if (typeof Proxy === 'function') {
+      var proxy = new Proxy(wrapper, {
+        get: function (target, prop, receiver) {
+          if (prop === 'worker') {
+            return worker;
+          }
+          return Reflect.get(...arguments);
+        }
+      });
+
+      return proxy;
+    } else {
+      wrapper.worker = worker;
+      return wrapper;
+    }
+  } `;
+
+
   // ?singleton mode: export an instance of the worker
   if (singleton === true) {
     return `
-      module.exports = require('comlink').wrap(require(${remainingRequest})());
-      ${options.module === false ? '' : 'module.exports.__esModule = true;'}
-    `.replace(/\n\s*/g, '');
+  ${wrapperBuilder}
+  module.exports = getWrappedWorker();
```

Comment:
In the singleton case it seems like there might not be a need for terminating the Worker since it's a side effect of module Instantiation. Thoughts?

Reply:
I don't use the singleton mode myself, but I thought it would be nice to have a consistent API and it allowed me to factor the code. 

There is one use case that I may think of. in the context of a SPA, one might want to terminate all singletons for various workers when the user logs out, to avoid potential access to data of the now gone user inside the worker once another user signs in to the app (ie. view it as a singleton for the lifetime of the user session as opposed to the lifetime of the browser tab/window). but as we say it in French: c'est tiré par les cheveux. (it's far fetched, literally "it's dragged by the hair")